install/kubernetes/cilium/templates/clustermesh-config/_helpers.tpl

```diff
@@ -3,7 +3,11 @@
 {{- $domain := index . 1 -}}
 {{- $override := index . 2 -}}
 {{- /* The parenthesis around $cluster.tls are required, since it can be null: https://stackoverflow.com/a/68807258 */}}
-{{- $prefix := ternary "common-" (printf "%s." $cluster.name) (or (ne $override "") (empty ($cluster.tls).cert) (empty ($cluster.tls).key)) -}}
+{{- $prefix := ternary "common-" (printf "%s." $cluster.name) (or (empty ($cluster.tls).cert) (empty ($cluster.tls).key)) -}}
+{{- /* KVStoreMesh is enabled */}}
+{{- if ne $override "" -}}
+{{- $prefix = "local-" -}}
+{{- end -}}
 
 endpoints:
```

Comment:
We had ~40 min session with Marco to help me understand templating magic here. 
For now, we decided to improve comments here a bit to make it more obvious what different conditions actually mean. 
For example `if ne $override ""` means KVStoreMesh is enabled **and** we are generating a secret for the agent (but not for the control plane).

Reply:
I've extended the comments. Hopefully that should clarify the logic here, which I agree being definitely convoluted.